process-manager.js

```diff
@@ -88,38 +93,46 @@ class ProcessManager {
 	release(process) {
 		process.release();
 	}
+
 	send() {
 		if (!this.processes.length) {
 			return Promise.resolve(this.receive.apply(this, arguments));
 		}
 
+		// Prevents the arguments object from leaking.
+		let args = [];
+		for (let i = 0; i < arguments.length; i++) {
+			args.push(arguments[i]);
+		}
```

Comment:
Ok, after messing with arguments for a while, and basically `.apply()` is the only way I could use the object here without leaking to some degree, though some ways leak more than others. The function's deoptimized no matter what I try doing. It's hard to tell if arguments truly leak or if they just dump a lot of objects on the heap without checking coredumps, but the GC could really use a little help anyways.

I'll fix what's still leaking, then hopefully it should be about done.